adapter/internal/oasparser/model/mgw_swagger.go

```diff
@@ -412,6 +414,37 @@ func (swagger *MgwSwagger) SetXWso2Extensions() error {
 	return nil
 }
 
+// SetRateLimitPoliciesForOperations assings rate limit policies to work with envoy rate limit service
+func (swagger *MgwSwagger) SetRateLimitPoliciesForOperations(apiYamlOperations []OperationYaml) {
+	if swagger.RateLimitLevel == "API" {
+		for _, resource := range swagger.resources {
+			for _, operation := range resource.methods {
+				operation.RateLimitPolicy = swagger.RateLimitPolicy
+			}
+		}
+	} else {
+		m := createOperationRateLimitDataMap(apiYamlOperations)
+		for _, resource := range swagger.resources {
+			for _, operation := range resource.methods {
+				key := resource.path + operation.method
+				if val, ok := m[key]; ok {
+					operation.RateLimitPolicy = val
+					logger.LoggerAPI.Debugf("Ratelimit policy %v found for the operation %v", val, key)
+				}
```

Comment:
Shall we print an error if an entry is missing?

Reply:
Would happen only if there was an inconsistency between the swagger and api.yaml. Which is highly unlikely. Just in case.